meta-client/src/main/java/org/apache/helix/metaclient/recipes/leaderelection/LeaderElectionClient.java

```diff
@@ -0,0 +1,180 @@
+   * When MetaClient is auto closed be cause of being disconnected and auto retry connection timed out, user
+   * will need to create a new MetaClient and a new LeaderElectionClient instance.
+   *
+   * @param metaClient metaClient object to be used.
+   */
+  public LeaderElectionClient(MetaClientInterface metaClient) {
+
+  }
+
+  /**
+   * Returns true if current participant is the current leadership.
+   */
+  public boolean isLeader(String leaderPath, String participant) {
+    return false;
+  }
+
+  /**
+   * Participants join a leader election group by calling the following API.
+   * The Leader Election client maintains and elect an active leader from the participant pool.
+   *
+   * @param leaderPath The path for leader election.
+   * @param participant The participant name to join pool.
+   * @return boolean indicating if the operation is succeeded.
+   */
+  public boolean joinLeaderElectionParticipantPool(String leaderPath, String participant) {
+    return false;
+  }
+
+  /**
+   * Participants join a leader election group by calling the following API.
+   * The Leader Election client maintains and elect an active leader from the participant pool.
+   *
+   * @param leaderPath The path for leader election.
+   * @param participant The participant name to join pool.
+   * @param userInfo Any additional information to associate with this participant.
+   * @return boolean indicating if the operation is succeeded.
+   */
+  public boolean joinLeaderElectionParticipantPool(String leaderPath, String participant,
+      Object userInfo) {
+    return false;
+  }
+
+  /**
+   * Any participant may exit the exitLeaderElectionParticipantPool by calling the API.
+   * If the participant is not the current leader, it leaves the pool and won't participant future
+   * leader election process.
+   * If the participant is the current leader, it leaves the pool and a new leader will be elected
+   * if there are other participants in the pool.
+   * Throws exception if the participant is not in the pool.
+   *
+   * @param leaderPath The path for leader election.
+   * @param participant The participant name to exit pool
+   * @return boolean indicating if the operation is succeeded.
+   *
+   * @throws RuntimeException If the participant did not join participant pool via this client. // TODO: define exp type
+   */
+  public boolean exitLeaderElectionParticipantPool(String leaderPath, String participant) {
+    return false;
+  }
+
+  /**
+   * Releases leadership for participant.
+   *
+   * @param leaderPath The path for leader election.
+   * @param participant The participant to release leadership.
+   *
+   * @throws RuntimeException if the leadership is not owned by this participant, or if the
+   *                          participant did not join participant pool via this client. // TODO: define exp type
+   */
+  public void relinquishLeader(String leaderPath, String participant) {
```

Comment:
does this mean anyone can call this and release leader

Reply:
```
* @throws RuntimeException if the leadership is not owned by this participant, or if the
*                          participant did not join participant pool via this client. // TODO: define exp type
```

Reply:
Had an offline discussion. 
One leadership election client is mapped to one host or participant. A participant may join multiple leader election group.  Updated the client API.